classes/class7/README.md

```diff
@@ -1 +1,152 @@
+`http://example.com/kb/index.php?cat=8&id=41` | `http://example.com/kb/8/41`
+`http://example.com/index.php?mod=profiles&id=193` | `http://example.com/profiles/193`
+*Source: http://en.wikipedia.org/wiki/Semantic_URL*
+
+Now if we combine everything we've learned about REST methods, semantic paths, and Wikipedia being a great resource for explaining things graphically, we get something like the following table.
+
+Resource | GET | PUT | POST | DELETE
+---------|-----|-----|------|-------
+**Collection URI**, such as http://example.com/resources/ | List the URIs and perhaps other details of the collection's members. | Replace the entire collection with another collection. | Create a new entry in the collection. The new entry's URI is assigned automatically and is usually returned by the operation. | Delete the entire collection.
+**Element URI**, such as http://example.com/resources/item17 | Retrieve a representation of the addressed member of the collection, expressed in an appropriate Internet media type. | Replace the addressed member of the collection, or if it doesn't exist, create it. | Not generally used. Treat the addressed member as a collection in its own right and create a new entry in it. | Delete the addressed member of the collection.
+
+*Source: http://en.wikipedia.org/wiki/Representational_state_transfer*
+
+###Versioning with Accept headers
+One thing you almost always see API providers doing in a way that is not semantic is specifying the version of the API to be used for the request by prefixing it to the resource route. Can you guess why this isn't semantic? That's right - the route is supposed to describe the resource, and our API version does not (or should not) do that. Thankfully, we know better than that, so where else can we put it? In the HTTP headers! Specifically the `Accept` header. The API version value would look something like `application/vnd.myapp.v1`, added to whatever values need to be sent in the `Accept` header. Then in our application, we can check it easily with the `accepts` package (install via npm):
+```js
+var accepts = require('accepts');
+
+app.use('/api*', function(req, res, next) {
+  var accept = accepts(req);
+  if (accept.types('application/vnd.myapp.v1')) {
+    next();
+  } else {
+    return res.json(400, {
+      message: "Incorrect API version"
+    });
+  }
+});
+```
+
+###Can I set up my own OAuth server?
+So you want to use OAuth to authenticate and authorize your API users, huh? Unfortunately that's _just_ outside the scope of this class, but know that if you do get to that point someday, there are some great packages available (on npm) that can help you get a basic setup running fairly quickly.
+
+_If this readme hasn't been enough for you and you want more REST, feel free to check out [this cool guide](http://www.infoq.com/articles/rest-introduction) on your own time._
+
+##Debugging
```

Comment:
I'd drop this line — we might get to debugging in class and we can edit the Readme if we do, maybe moving in some stuff from last class.